src/main/java/tech/allegro/wildsnake/rest/ProductsEndPoint.java

```diff
@@ -60,13 +56,15 @@ public void updateProduct(
 
     @RequestMapping(
             method = RequestMethod.PUT,
-            consumes = MediaType.APPLICATION_JSON_VALUE
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            value = "//{productCategoryName}"
     )
     @ResponseStatus(HttpStatus.NO_CONTENT)
     public void addProduct(
-            @RequestBody ProductDomain productDomain
+            @RequestBody ProductDomain productDomain, @PathVariable("productCategoryName") String
```

Comment:
Can product category id or name be included in ProductDomain object? It will simplify the controller and the way people communicate with api?.